### ISSUE TITLE
Counting instructions using Stalker

### ISSUE THREAD

**Issue by monosource** [frida/frida-python#94]
I would like to contribute to Frida by instrumenting Linux binaries and doing write-ups. I think it's a brilliant project as you don't need to recompile the instrumentation code like in Intel Pin. I think it would be great to let people who don't necessarily instrument iOS or Android know about Frida.

I am currently trying to count the number of instructions executed by a given ELF32 binary using Stalker. My code is the following

```python
import frida
import sys

stalker="""
    Stalker.follow(Process.getCurrentThreadId(),
    {
        events: {call: true, exec: true, ret: true},
        onCallSummary: function (summary) {
            send(summary);
        }
    });
"""

pid = frida.spawn(['./stalkme'])
process = frida.attach(pid)

def on_message(message, data):
    print message
    print data

script = process.create_script(stalker)  
script.on('message', on_message)
script.load()
sys.stdin.read() # Don't let the parent exit
```

This, however, prints the following output indefinitely:
```
{u'type': u'send', u'payload': {}}
None
```

I've used the console log to look at the **summary** argument, and I get a nice hexdump of it:
```
           0  1  2  3  4  5  6  7  8  9  A  B  C  D  E  F  0123456789ABCDEF
00000000  04 00 00 00 67 e3 97 f5 00 00 00 00 00 00 00 00  ....g...........
00000010  04 00 00 00 7c e3 97 f5 00 00 00 00 00 00 00 00  ....|...........
00000020  04 00 00 00 7f e3 97 f5 00 00 00 00 00 00 00 00  ................
00000030  04 00 00 00 82 e3 97 f5 00 00 00 00 00 00 00 00  ................
00000040  04 00 00 00 89 e3 97 f5 00 00 00 00 00 00 00 00  ................
00000050  04 00 00 00 8b e3 97 f5 00 00 00 00 00 00 00 00  ................
00000060  04 00 00 00 8d e3 97 f5 00 00 00 00 00 00 00 00  ................
-----------------------------------[SNIPPED]-------------------------------
```

As far as I understand from the [API](http://www.frida.re/docs/javascript-api/#stalker), this is a list of [GumEvent structs](https://github.com/frida/frida-gum/blob/master/gum/gumevent.h). If I want to go any deeper to see what instructions are executed, I have to parse the struct, retrieve the pointers, read at the address, disassemble the instruction.

But going back to instruction counting, I should just have a global counter in my script which I increment in **on_message**. The **sys.stdin.read()** at the end makes it so that execution never stops. If I comment it out, then the counter is never incremented since the parent exits.

TLDR: How do I properly use Stalker to count the number of instructions executed by an ELF binary?

**Comment by oleavr** [comment]
That would be awesome! To do that, you have to tell it you want `exec` events, and then you can look at the accumulated size of events and divide by size of the `GumEvent` struct.

Cheers!
